packages-content-model/roosterjs-content-model-editor/lib/modelApi/common/mergeModel.ts

```diff
@@ -131,6 +134,16 @@ function mergeParagraph(
 
     if (newPara.decorator) {
         newParagraph.decorator = { ...newPara.decorator };
+        if (HeadingTags.indexOf(newParagraph.decorator.tagName) > -1) {
+            const keys: (keyof ContentModelSegmentFormat)[] = getObjectKeys(
+                newParagraph.decorator.format
+            );
+            keys.forEach(key => {
+                if (newParagraph.segmentFormat?.[key]) {
```

Comment:
when the value is false, or 0, this will also return false. Is that expected?

Reply:
Good catch. Fixed in latest commit